src/models/base.es6.js

```diff
@@ -1,59 +1,41 @@
-import { EventEmitter } from 'events';
+const THING_ID_REGEX = new RegExp('t\\d_[0-9a-z]+', 'i');
 
 class Base {
-  constructor (props={}, opts={}) {
+  constructor (props={}) {
     this.props = {};
-    this.validators = opts.validators || [];
-    this.formatters = opts.formatters || [];
-    this.emitter = opts.emitter || new EventEmitter();
 
-    for (var p in props) {
+    for (let p in props) {
       this.props[p] = props[p];
     }
   }
 
+  get _type () {
+    return this.constructor.name;
```

Comment:
This seems like it wouldn't work across minifying, do we (/ other js bundlers in general) have something a prop for this during minifying?